Packs/AtlassianConfluenceCloud/Integrations/AtlassianConfluenceCloud/AtlassianConfluenceCloud.py

```diff
@@ -139,14 +148,95 @@ def exception_handler(response: requests.models.Response):
+        )
+
 
 ''' HELPER FUNCTIONS '''
 
 
+def get_last_run() -> dict:
+    """
+    Retrieve the last run data from the cache or Demisto's getLastRun.
+
+    Returns:
+        dict: The last run data.
+    """
+    global _last_run_cache
+    if _last_run_cache is None:
+        _last_run_cache = demisto.getLastRun()
+    return _last_run_cache
+
+
+def set_last_run(last_run: dict):
+    """
+    Set the last run data in both the cache and Demisto's setLastRun.
+
+    Args:
+        last_run (dict): The last run data to set.
+    """
+    global _last_run_cache
+    _last_run_cache = last_run
+    demisto.setLastRun(last_run)
+
+
+def run_fetch_mechanism(client: Client, next_link: str | None, last_run: dict[str, Any], page_size: int) -> dict[str, Any]:
+    """
+    Retrieve events from the Atlassian Confluence Cloud API according to the following logic:
+        - If next_link is not provided, it searches for events within the specified date range.
+        - If next_link is provided, it fetches the next page of results.
+    The mechanism ensures that while we have a next_link, we continue fetching pages until there are no more pages to fetch.
+    Args:
+        client (Client): The client instance for making API requests.
+        next_link (str | None): The link for the next page of results, if any.
+        last_run (dict[str, Any]): The dictionary containing the last run information, such as the end date.
+        page_size (int): The desired page size for the search.
+    Returns:
+        dict[str, Any]: The API response containing the fetched events.
+    """
+    if not next_link:
+        end_date = int((time.time() - 5) * 1000)
+        last_end_date = last_run.get('end_date', 0)
+        start_date = last_end_date + 1 if last_end_date else end_date - ONE_MINUTE_IN_MILL_SECONDS
+        if end_date < start_date or end_date - start_date < 30000:
+            # end run
+            pass
+        demisto.debug(f'searching events with start date: {start_date}, end date: {end_date} and page size: {page_size}')
+        response = client.search_events(limit=page_size, start_date=str(start_date), end_date=str(end_date))
+        demisto.debug(f'Found {response["size"]} events between {start_date} and {end_date}')
+        demisto.debug(json.dumps(response, indent=4))
+
+    else:
+        demisto.debug(f'searching events with next_link: {next_link} and page size: {AUDIT_FETCH_PAGE_SIZE}')
+        response = client.search_events(limit=page_size, next_link=next_link)
+        demisto.debug(f'Found {response["size"]} events in the current page')
+        demisto.debug(json.dumps(response, indent=4))
+
+    return response
+
+
+def add_time_to_events(events: list[dict] | None):
+    """
+    Adds the _time key to the events.
+    Args:
+        events: List[Dict] - list of events to add the _time key to.
+    Returns:
+        list: The events with the _time key.
+    """
+    if events:
```

Comment:
this check is done outside, can be removed

Reply:
done